playbooks/openshift-master/scaleup.yml

```diff
@@ -1,22 +1,43 @@
 ---
 - import_playbook: ../init/evaluate_groups.yml
 
-- name: Ensure there are new_masters or new_nodes
+- name: Ensure there are new_masters and new_nodes
   hosts: localhost
   connection: local
   gather_facts: no
   tasks:
   - fail:
+      # new_masters must be part of new_nodes as well; otherwise if new_nodes
+      # is not present, oo_nodes_to_config will contain all existing nodes.
       msg: >
-        Detected no new_masters or no new_nodes in inventory. Please
-        add hosts to the new_masters and new_nodes host groups to add
-        masters.
-    when:
-    - g_new_master_hosts | default([]) | length == 0
-    - g_new_node_hosts | default([]) | length == 0
+        Detected no new_masters and/or no new_nodes in inventory. New
+        masters must be part of both new_masters and new_nodes groups.
+        If you are adding just new_nodes, use the
+        playbooks/openshift-node/scaleup.yml play.
+    when: >
+      g_new_master_hosts | default([]) | length == 0
+      or g_new_node_hosts | default([]) | length == 0
 
-# Need a better way to do the above check for node without
-# running evaluate_groups and init/main.yml
-- import_playbook: ../init/main.yml
+- name: Ensure there are new_masters and new_nodes
+  hosts: oo_masters_to_config
+  connection: local
+  gather_facts: no
+  tasks:
+  - fail:
+      # new_masters must be part of new_nodes as well;
+      msg: >
+        Each host in new_masters must also appear in new_nodes
```

Comment:
Not sure why wouldn't `scaleup` disallow scaling up dedicated masters

Reply:
Do we actually run the node scaleup playbook to complete the node specific tasks or do we need to document the need to run node scaleup too?

Reply:
@sdodson this openshift-master/private/scaleup.yml and openshift-node/private/scaleup.yml both call openshift-node/private/config.yml.  There is no need to run openshift-node/scaleup in addition to this play if you're adding both masters and nodes.  The nodes were already being configured at the same time as masters, according to our plays.

If one is adding nodes and 0 masters, then they can run openshift-node/scaleup.yml.

Reply:
Sounds good to me, thanks.

Reply:
>in 3.9 masters will actually be expected to run the console pods.

Oh, I see, makes sense.

> Do we actually run the node scaleup playbook to complete the node specific tasks

Yes, the node is being setup there